File: Models/Simplace_Soil_Temperature/src/cs/Simplace_Soil_Temperature/SoilTemperatureExogenous.cs
using System;
using System.Collections.Generic;

public class SoilTemperatureExogenous 
{
    private double _iAirTemperatureMax;
    private double _iTempMax;
    private double _iAirTemperatureMin;
    private double _iTempMin;
    private double _iGlobalSolarRadiation;
    private double _iRadiation;
    private double _iRAIN;
    private double _iCropResidues;
    private double _iPotentialSoilEvaporation;
    private double _iLeafAreaIndex;
    private double _iSoilWaterContent;
    
    /// <summary>
    /// Constructor of the SoilTemperatureExogenous component")
    /// </summary>  
    public SoilTemperatureExogenous() { }
    
    
    public SoilTemperatureExogenous(SoilTemperatureExogenous toCopy, bool copyAll) // copy constructor 
    {
        if (copyAll)
        {
    
            iAirTemperatureMax = toCopy.iAirTemperatureMax;
            iTempMax = toCopy.iTempMax;
            iAirTemperatureMin = toCopy.iAirTemperatureMin;
            iTempMin = toCopy.iTempMin;
            iGlobalSolarRadiation = toCopy.iGlobalSolarRadiation;
            iRadiation = toCopy.iRadiation;
            iRAIN = toCopy.iRAIN;
            iCropResidues = toCopy.iCropResidues;
            iPotentialSoilEvaporation = toCopy.iPotentialSoilEvaporation;
            iLeafAreaIndex = toCopy.iLeafAreaIndex;
            iSoilWaterContent = toCopy.iSoilWaterContent;
        }
    }
    public double iAirTemperatureMax
    {
        get { return this._iAirTemperatureMax; }
        set { this._iAirTemperatureMax= value; } 
    }
    public double iTempMax
    {
        get { return this._iTempMax; }
        set { this._iTempMax= value; } 
    }
    public double iAirTemperatureMin
    {
        get { return this._iAirTemperatureMin; }
        set { this._iAirTemperatureMin= value; } 
    }
    public double iTempMin
    {
        get { return this._iTempMin; }
        set { this._iTempMin= value; } 
    }
    public double iGlobalSolarRadiation
    {
        get { return this._iGlobalSolarRadiation; }
        set { this._iGlobalSolarRadiation= value; } 
    }
    public double iRadiation
    {
        get { return this._iRadiation; }
        set { this._iRadiation= value; } 
    }
    public double iRAIN
    {
        get { return this._iRAIN; }
        set { this._iRAIN= value; } 
    }
    public double iCropResidues
    {
        get { return this._iCropResidues; }
        set { this._iCropResidues= value; } 
    }
    public double iPotentialSoilEvaporation
    {
        get { return this._iPotentialSoilEvaporation; }
        set { this._iPotentialSoilEvaporation= value; } 
    }
    public double iLeafAreaIndex
    {
        get { return this._iLeafAreaIndex; }
        set { this._iLeafAreaIndex= value; } 
    }
    public double iSoilWaterContent
    {
        get { return this._iSoilWaterContent; }
        set { this._iSoilWaterContent= value; } 
    }
}

File: Models/Simplace_Soil_Temperature/src/cpp/Simplace_Soil_Temperature/SoilTemperatureExogenous.cpp
#include "SoilTemperatureExogenous.h"
using namespace Simplace_Soil_Temperature;

SoilTemperatureExogenous::SoilTemperatureExogenous() {}

double SoilTemperatureExogenous::getiAirTemperatureMax() { return this->iAirTemperatureMax; }
double SoilTemperatureExogenous::getiTempMax() { return this->iTempMax; }
double SoilTemperatureExogenous::getiAirTemperatureMin() { return this->iAirTemperatureMin; }
double SoilTemperatureExogenous::getiTempMin() { return this->iTempMin; }
double SoilTemperatureExogenous::getiGlobalSolarRadiation() { return this->iGlobalSolarRadiation; }
double SoilTemperatureExogenous::getiRadiation() { return this->iRadiation; }
double SoilTemperatureExogenous::getiRAIN() { return this->iRAIN; }
double SoilTemperatureExogenous::getiCropResidues() { return this->iCropResidues; }
double SoilTemperatureExogenous::getiPotentialSoilEvaporation() { return this->iPotentialSoilEvaporation; }
double SoilTemperatureExogenous::getiLeafAreaIndex() { return this->iLeafAreaIndex; }
double SoilTemperatureExogenous::getiSoilWaterContent() { return this->iSoilWaterContent; }

void SoilTemperatureExogenous::setiAirTemperatureMax(double _iAirTemperatureMax) { this->iAirTemperatureMax = _iAirTemperatureMax; }
void SoilTemperatureExogenous::setiTempMax(double _iTempMax) { this->iTempMax = _iTempMax; }
void SoilTemperatureExogenous::setiAirTemperatureMin(double _iAirTemperatureMin) { this->iAirTemperatureMin = _iAirTemperatureMin; }
void SoilTemperatureExogenous::setiTempMin(double _iTempMin) { this->iTempMin = _iTempMin; }
void SoilTemperatureExogenous::setiGlobalSolarRadiation(double _iGlobalSolarRadiation) { this->iGlobalSolarRadiation = _iGlobalSolarRadiation; }
void SoilTemperatureExogenous::setiRadiation(double _iRadiation) { this->iRadiation = _iRadiation; }
void SoilTemperatureExogenous::setiRAIN(double _iRAIN) { this->iRAIN = _iRAIN; }
void SoilTemperatureExogenous::setiCropResidues(double _iCropResidues) { this->iCropResidues = _iCropResidues; }
void SoilTemperatureExogenous::setiPotentialSoilEvaporation(double _iPotentialSoilEvaporation) { this->iPotentialSoilEvaporation = _iPotentialSoilEvaporation; }
void SoilTemperatureExogenous::setiLeafAreaIndex(double _iLeafAreaIndex) { this->iLeafAreaIndex = _iLeafAreaIndex; }
void SoilTemperatureExogenous::setiSoilWaterContent(double _iSoilWaterContent) { this->iSoilWaterContent = _iSoilWaterContent; }